### DISCUSSION TITLE
"Server closed connection during identification exchange" for sshj version 0.33.0 with kernel version "Linux 5.4.0-91-generic x86_64"

### DISCUSSION THREAD

**Discussion by nandroy** [hierynomus/sshj#788] [Q&A]
Facing exception "Server closed connection during identification exchange" for sshj version 0.33.0 with kernel version "Linux 5.4.0-91-generic x86_64" for the following code segment of PortForwarding.

SSHClient clientSsh;
Parameters params;
ServerSocket serversocket;
LocalPortForwarder localPortForwarderObj;

this.localPortForwarderObj = this.clientSsh.newLocalPortForwarder(this.params, this.serversocket);
this.localPortForwarderObj.listen(); // this line gives exception

Which was working fine when kernel version was "Linux 4.15.0-142-generic x86_64".

What changes need to be done for port forwarding in latest version of sshj 0.33.0 ?

**Comment by hierynomus** [comment]
You would need to add more debug logging to see why the server closes the connection. Are you using BouncyCastle? If not, the server probably rejects insecure/old algorithms.
